pytket/extensions/qiskit/backends/ibm.py

```diff
@@ -525,11 +533,14 @@ def process_circuits(
 
                 qcs, ppcirc_strs = [], []
                 for tkc in batch_chunk:
+                    tkc1 = tkc.copy()
+                    # Flatten bits to default register in lexicographic order:
```

Comment:
This solves the issue raised in [this comment](https://github.com/CQCL/pytket-qiskit/pull/335#discussion_r1613566199). There is still an issue with circuits having non-default classical register names, but this is an independent pre-existing issue and is not addressed in this PR.

Reply:
Just to make sure I understand: this does actually all work if `pytket-qiskit` is used to submit and retrieve the circuit no matter what the classical registers are named. Of course if one separately submits a circuit with non-default classical register names and tries to retrieve it with `pytket-qiskit` then there is an error... but that is indeed a different issue!

Reply:
Yes, it works _unless_ you then want to call `get_counts()` on the result with specific bits, using the `cbits` parameter to `get_counts`. In that case it will throw an error -- but that was already the case (separate issue).

Reply:
Thanks that makes sense!